en_US/data-integration/data-bridge-tdengine.md

```diff
@@ -0,0 +1,162 @@
+# TDengine
+
+EMQX supports integration with TDengine so you can save client messages and events to TDengine, or use events to trigger the update or removal of data to record the online status or online/offline of clients.
+
+## Prerequisites
+
+- Knowledge about EMQX data integration [rules](./rules.md)
+- Knowledge about [data bridge](./data-bridges.md)
+
+## Features list
+
+- [Connection pool](./data-bridges.md#Connection pool)
+- [Async mode](./data-bridges.md#Async mode)
+- [Batch mode](./data-bridges.md#Batch mode)
+- [Buffer queue](./data-bridges.md#Buffer queue)
+- [SQL preprocessing](./data-bridges.md#Prepared statement)
+
+## [Configuration parameters](#Configuration)
+<!-- TODO LIKN TO THE CONFIG doc。 -->
+
+## Quick starts
+
+### Install TDengine
+
+Install TDengine via Docker, and then run the docker image. 
+
+```bash
+# To start the TDengine docker image 
+docker run --name TDengine -p 6041:6041 tdengine/tdengine
+
+# Access the container
+docker exec -it TDengine bash
+
+# Locate the TDengine server in the container
+taos
+
+# Create and then select the database
+CREATE DATABASE mqtt;
+
+use mqtt;
+```
+
+### Connect to TDengine
+
+We will create 2 data bridges to TDengine for messages storage and event records respectively. 
+
+#### [Messages storage](#Storage)
+
+1. Go to EMQX Dashboard, click **Data Integration** -> **Data Bridge**.
+2. Click **Create** on the top right corner of the page.
+3. In the **Create Data Bridge** page, click to select **TDengine**, and then click **Next**.
+4. Input a name for the data bridge. Note: It should be a combination of upper/lower case letters and numbers.
+5. Input the connection information. Input **127.0.0.1:6041** as the **Server Host**,  **mqtt** as the **Database Name**, **root** as the **Username**, and **taosdata** as the **Password**.
+6. Configure the **SQL Template**. Use the SQL statements below to insert data. Note: This is a preprocessed SQL, so the fields should not be enclosed in quotation marks, and do not write a semicolon at the end of the statements. 
+
+  ```sql
+    INSERT INTO mqtt.t_mqtt_msg(ts, msgid, mqtt_topic, qos, payload, arrived) 
+      VALUES (${ts}, ${id}, ${topic}, ${qos}, ${payload}, ${timestamp})
+  ```
+
+Before creating the above data bridge, please use the following SQL statements to create data table `t_mqtt_msg` in TDengine database for storing the client ID, topic, payload, and creating time of every message. 
+
+  ```sql
+    CREATE TABLE t_mqtt_msg (
+      ts timestamp,
+      msgid NCHAR(64),
+      mqtt_topic NCHAR(255),
+      qos TINYINT,
+      payload BINARY(1024),
+      arrived timestamp
+    );
+  ```
+
+7. Advanced settings (optional):  Choose whether to use sync or async query mode as needed. For details, see [Configuration parameters](#Configuration).
```

Comment:
For details, see [Configuration parameters](#Configuration). let's hide this part for the moment, it has caused some confusions among our users.